packages/wrangler/src/deployment-bundle/module-collection.ts

```diff
@@ -251,6 +252,42 @@ export function createModuleCollector(props: {
 								// it to `esbuild` to bundle it.
 								if (isJavaScriptModuleRule(rule)) return;
 
+								// Check if this file is possibly from an npm package
+								// and if so, validate the import against the package.json exports
+								// and resolve the file path to the correct file.
+								if (args.path.includes("/") && !args.path.startsWith("./")) {
```

Comment:
This should cover more cases https://nodejs.org/api/path.html#pathisabsolutepath
```suggestion
								if (args.path.includes("/") && !path.isAbsolute(args.path)) {
```

Reply:
Oh I misread it, nevermind